Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.9
```

Comment:
3.9?

Reply:
We should be good on 3.10 (or even 3.11), we're already running tests on both. We should also be okay using the slim variants. 

I decided those changes should go into future PRs, but happy to make them here if you think it's needed here.